src/services/expense.service.js

```diff
@@ -0,0 +1,80 @@
+const { Expense } = require('../models/Expense.model');
+const { Op } = require('sequelize');
+
+const getAll = async (category, userId, from, to) => {
+  const whereClause = {};
+
+  if (category) {
+    whereClause.category = category;
+  }
+
+  if (userId) {
+    whereClause.userId = userId;
+  }
+
+  if (from || to) {
+    whereClause.spentAt = {};
+
+    if (from) {
+      whereClause.spentAt[Op.gte] = new Date(from);
```

Comment:
Ensure that the `from` date is valid before using it to create a new Date object. Invalid dates can lead to unexpected behavior. Consider adding validation to check if `from` is a valid date string.